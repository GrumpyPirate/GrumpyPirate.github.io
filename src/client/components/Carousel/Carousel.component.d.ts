import { ChildrenProps } from 'types';

interface CarouselProps extends ChildrenProps {
  showStatus: boolean;
  showThumbs: boolean;
  infiniteLoop: boolean;
  useKeyboardArrows: boolean;
  transitionTime: number;
  dynamicHeight: boolean;
  showArrows: false;
  emulateTouch: boolean;
  swipeScrollTolerance: number;
}

export { CarouselProps };
