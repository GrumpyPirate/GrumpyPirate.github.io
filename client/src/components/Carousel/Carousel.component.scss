@import 'core';

:global {
  .carousel {
    position: relative;
    width: 100%;

    * {
      box-sizing: border-box;
      margin: 0;
      padding: 0;
    }

    button {
      background: none;
      border: 0;
      outline: 0;
    }

    img {
      pointer-events: none;
    }

    .carousel {
      position: relative;
    }

    .control-arrow {
      font-size: 18px;
      margin-top: -13px;
      top: 50%;
    }

    .thumbs-wrapper {
      margin: 20px;
      overflow: hidden;
    }

    .thumbs {
      list-style: none;
      position: relative;
      transform: translate3d(0, 0, 0);
      transition: all 0.15s ease-in;
      white-space: nowrap;
    }

    .thumb {
      border: 3px solid #fff;
      display: inline-block;
      margin-right: 6px;
      overflow: hidden;
      padding: 2px;
      transition: border 0.15s ease-in;
      white-space: nowrap;
      width: 80px;

      &.selected,
      &:hover {
        border: 3px solid #333;
        padding: 2px;
      }

      img {
        vertical-align: top;
      }
    }

    .control-dots {
      margin: 0.5rem 0;
      text-align: center;

      .dot {
        background: palette('theme-accent-light');
        border-radius: 50%;
        cursor: pointer;
        display: inline-block;
        height: 8px;
        margin: 0 8px;
        opacity: 0.5;
        transition-duration: $anim-dur;
        transition-property: border, opacity, transform, background-color;
        width: 8px;

        &:hover,
        &.selected {
          background-color: transparent;
          border: solid 1px palette('theme-accent-light');
          transform: scale(2);
        }

        &.selected {
          opacity: 1;
        }
      }
    }
  }

  .carousel .control-arrow,
  .carousel.carousel-slider .control-arrow {
    background: none;
    border: 0;
    cursor: pointer;
    font-size: 32px;
    opacity: 0.4;
    position: absolute;
    top: 20px;
    transition: all 0.25s ease-in;
    z-index: 2;

    .control-arrow {
      &:hover {
        opacity: 1;
      }

      &::before {
        border-bottom: 8px solid transparent;
        border-top: 8px solid transparent;
        content: '';
        display: inline-block;
        margin: 0 5px;
      }
    }
  }

  .carousel .control-disabled.control-arrow {
    cursor: inherit;
    display: none;
    opacity: 0;
  }

  .carousel .control-prev.control-arrow {
    left: 0;
  }

  .carousel .control-prev.control-arrow::before {
    border-right: 8px solid #fff;
  }

  .carousel .control-next.control-arrow {
    right: 0;
  }

  .carousel .control-next.control-arrow::before {
    border-left: 8px solid #fff;
  }

  .carousel.carousel-slider {
    margin: 0;
    overflow: hidden;
    position: relative;
  }

  .carousel.carousel-slider .control-arrow {
    bottom: 0;
    color: #fff;
    font-size: 26px;
    margin-top: 0;
    padding: 5px;
    top: 0;
  }

  .carousel.carousel-slider .control-arrow:hover {
    background: rgba(0, 0, 0, 0.2);
  }

  .carousel .slider-wrapper {
    margin: auto;
    overflow: hidden;
    transition: height 0.15s ease-in;
    width: 100%;
  }

  .carousel .slider-wrapper.axis-horizontal .slider {
    display: flex;
  }

  .carousel .slider-wrapper.axis-horizontal .slider .slide {
    flex-direction: column;
    flex-flow: column;
  }

  .carousel .slider-wrapper.axis-vertical {
    display: flex;
  }

  .carousel .slider-wrapper.axis-vertical .slider {
    flex-direction: column;
  }

  .carousel .slider {
    list-style: none;
    position: relative;
    width: 100%;
  }

  .carousel .slider.animated {
    transition: all .35s ease-in-out;
  }

  .carousel .slide {
    margin: 0;
    min-width: 100%;
    position: relative;
    text-align: center;
  }

  .carousel .slide img {
    border: 0;
    vertical-align: top;
  }

  .carousel .slide iframe {
    border: 0;
    display: inline-block;
    margin: 0 40px 40px;
    width: calc(100% - 80px);
  }

  .carousel .slide .legend {
    background: #000;
    border-radius: 10px;
    bottom: 40px;
    color: #fff;
    font-size: 12px;
    left: 50%;
    margin-left: -45%;
    opacity: 0.25;
    padding: 10px;
    position: absolute;
    text-align: center;
    transition: all 0.5s ease-in-out;
    transition: opacity 0.35s ease-in-out;
    width: 90%;
  }

  .carousel .carousel-status {
    color: #fff;
    font-size: 10px;
    padding: 5px;
    position: absolute;
    right: 0;
    text-shadow: 1px 1px 1px rgba(0, 0, 0, 0.9);
    top: 0;
  }

  .carousel:hover .slide .legend {
    opacity: 1;
  }
}
