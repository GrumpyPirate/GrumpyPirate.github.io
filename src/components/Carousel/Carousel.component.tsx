import React, { FunctionComponent } from 'react';
import { Carousel as ReactCarousel } from 'react-responsive-carousel';

import { CarouselProps } from './Carousel.component.types';

import './Carousel.component.scss';

const Carousel: FunctionComponent<CarouselProps> = ({ children, ...rest }) => (
  <ReactCarousel {...rest}>{children}</ReactCarousel>
);

export default Carousel;
