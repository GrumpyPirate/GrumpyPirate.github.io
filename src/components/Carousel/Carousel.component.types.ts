import { ChildrenProps } from 'types/common';

export interface CarouselProps extends ChildrenProps {
  showStatus: boolean;
  showThumbs: boolean;
  infiniteLoop: boolean;
  useKeyboardArrows: boolean;
  transitionTime: number;
  dynamicHeight: boolean;
  showArrows: false;
  emulateTouch: boolean;
  swipeScrollTolerance: number;
}
