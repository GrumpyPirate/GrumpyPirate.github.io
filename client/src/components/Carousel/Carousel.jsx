import React from 'react';
import PropTypes from 'prop-types';

import './Carousel.scss';

import { Carousel as ReactCarousel } from 'react-responsive-carousel';

const Carousel = ({ children, ...rest }) => <ReactCarousel {...rest}>{children}</ReactCarousel>;

// PropTypes
Carousel.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Carousel;
