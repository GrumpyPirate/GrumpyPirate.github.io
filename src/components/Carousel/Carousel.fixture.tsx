import React from 'react';

import { CarouselProps } from './Carousel.types';

const carouselProps: CarouselProps = {
  children: [<div>Test</div>],
  showStatus: false,
  showThumbs: false,
  infiniteLoop: true,
  useKeyboardArrows: true,
  transitionTime: 500,
  dynamicHeight: true,
  showArrows: false,
  emulateTouch: true,
  swipeScrollTolerance: 75,
};

export default carouselProps;
