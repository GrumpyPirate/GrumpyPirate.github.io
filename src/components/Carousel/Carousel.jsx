// React
import React from 'react'
import PropTypes from 'prop-types'

// SCSS
import './Carousel.scss'

// Vendor
import { Carousel as ReactCarousel } from 'react-responsive-carousel'

// Define Carousel
const Carousel = (props) => {
  return (
    <ReactCarousel {...props}>
      {props.children}
    </ReactCarousel>
  )
}

Carousel.propTypes = {
  children: PropTypes.oneOfType(
    [
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node
    ]
  )
}

export default Carousel
