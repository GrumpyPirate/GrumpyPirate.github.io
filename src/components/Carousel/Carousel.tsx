import React, { FunctionComponent } from 'react';
import { Carousel as ReactCarousel } from 'react-responsive-carousel';

import { CarouselProps } from './Carousel.types';

import './Carousel.scss';

const Carousel: FunctionComponent<CarouselProps> = ({ children, ...rest }) => (
  <ReactCarousel {...rest}>{children}</ReactCarousel>
);

export default Carousel;
