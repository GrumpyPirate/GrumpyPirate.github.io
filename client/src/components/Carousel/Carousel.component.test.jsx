import React from 'react';
import { shallow } from 'enzyme';

import Carousel from './Carousel.component';

import fixture from './Carousel.component.fixture';

describe('Components', () => {
  describe('Carousel', () => {
    const props = {
      ...fixture,
    };

    it('should be defined', () => {
      expect(Carousel).toBeDefined();
    });

    it('should be a function', () => {
      expect(typeof Carousel).toBe('function');
    });

    it('should render as expected, without crashing', () => {
      const wrapper = shallow(<Carousel {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
