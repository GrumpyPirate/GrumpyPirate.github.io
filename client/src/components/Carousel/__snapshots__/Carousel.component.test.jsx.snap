// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Carousel should render as expected, without crashing 1`] = `ShallowWrapper {}`;
